lib/Markdown/Perl/BlockParser.pm

```diff
@@ -137,8 +137,15 @@ sub process {
   # Done at a later stage, as escaped characters don’t have their Markdown
   # meaning, we need a way to represent that.
 
-  # Note: for now, nothing is done with the extracted metadata.
-  $this->_parse_yaml_metadata() if $this->get_parse_file_metadata eq 'yaml';
+  if($this->get_parse_file_metadata eq 'yaml') {
+    my $hook_result = eval {
```

Comment:
My suggestion was to not even try to catch the exception and just let it be propagated to the initial caller, so that they can’t miss that the document has not been processed.